### PR TITLE
Maps API Implementation: Week 4

### PR DESCRIPTION
**Pull request by sunr1**
Implemented the Maps API

Overview of the map that has pins where the images from the hobby icons were taken:
image

Each pin has a description once clicked: 
image

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -50,3 +51,130 @@ function appendCommentElement(txt, commentDiv) {
...
+        elementType: 'labels.text.fill',
+        stylers: [{color: '#515c6d'}]
+      },
+      {
+        featureType: 'water',
+        elementType: 'labels.text.stroke',
+        stylers: [{color: '#17263c'}]
+      }
+    ]
+  });
```
Comment (kzweller) [inline portfolio/src/main/webapp/script.js]: This is a lot. I would either:

a) Make a separate JS file for this method.
b) save these as JSON files and use fs to parse them (https://medium.com/@osiolabs/read-write-json-files-with-node-js-92d03cc82824)

I'm inclined to say the latter, but I don't truly know what would be involve.

**Review by kzweller — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -85,15 +88,12 @@ <h4 class="category">-INTERESTS & HOBBIES-</h4>
...
-
-        <a href="https://github.com/sunr1" target="_blank"><img src="https://img.icons8.com/ios-filled/50/000000/github.png" alt="images/linkedin.png"/></a>
-        <a href="https://linkedin.com/in/rebecca-sun/" target="_blank"><img src="https://img.icons8.com/color/48/000000/linkedin.png" alt="images/github.png"/></a>
-        <a href="mailto:rebeccasun@google.com" target="_blank"><img src="https://img.icons8.com/fluent/48/000000/gmail.png" alt="images/gmail.png"/></a>
     </div>
 
     <footer>
         <p id="contact">Get in touch: <a href="mailto:rebeccasun@google.com" target="_blank"> rebeccasun@google.com</a></p>
         <p id="credits">Created and designed by Rebecca Sun</p>
+        <button onclick="confetti()"></button>
```
Comment (tambchop) [inline portfolio/src/main/webapp/index.html]: Is this button intentional?  Like an easter egg haha?  The confetti effect is super cool!  But it shows up as a tiny sliver across the very bottom of the webpage.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -50,3 +51,130 @@ function appendCommentElement(txt, commentDiv) {
...
+  var culvInfoWindow = new google.maps.InfoWindow({content: culvString});
+  var nashInfoWindow = new google.maps.InfoWindow({content: nashString});
+
+  setMarker(kcMarker, kcInfoWindow);
+  setMarker(culvMarker, culvInfoWindow);
+  setMarker(nashMarker, nashInfoWindow);
+}
+
+function setMarker(marker, infoWindow) {
+  marker.setMap;
```
Comment (tambchop) [inline portfolio/src/main/webapp/script.js]: What is this line doing?  setMap is a method, so it should take some arguments, right?  Should it be marker.setMap(map)?  The map seems to be working when I interact with it, so I'm not sure what the deal is... maybe this line is unnecessary?

Comment (tambchop) [reply portfolio/src/main/webapp/script.js]: Oh also I see when you're creating the markers using "new google.maps.Marker(..." above, one of the fields you are passing in is "map: map", so I think the markers already have their map value set.  So I think that this line here can likely be deleted =)

File: portfolio/src/main/webapp/index.html (hunk 2 above)
Comment (sunr1) [reply portfolio/src/main/webapp/index.html]: I made it a little bigger haha thanks for pointing that out

**Review by tambchop — APPROVED** [review]

File: portfolio/src/main/webapp/map.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  const kcString = '<h4>Kansas City, Kansas</h4>' +
+      '<p>My hometown</p>';
+  const culvString = '<h4>Culver, Indiana</h4>' +
+      '<p>Where I attended a boarding high school for four years</p>';
+  const nashString = '<h4>Nashville, Tennessee</h4>' +
+      '<p>Where I am currently attending college</p>';
+
+  var kcInfoWindow = new google.maps.InfoWindow({content: kcString});
+  var culvInfoWindow = new google.maps.InfoWindow({content: culvString});
+  var nashInfoWindow = new google.maps.InfoWindow({content: nashString});
```
Comment (tambchop) [inline portfolio/src/main/webapp/map.js]: These should be const instead of var.

**Review by kzweller — APPROVED** [review]

**Pull request merged**